xbmc/filesystem/DirectoryFactory.h

```diff
@@ -9,6 +9,7 @@
 #pragma once
 
 #include "IDirectory.h"
+#include "addons/addoninfo/AddonType.h"
```

Comment:
Same here: common code/base classes should be agnostic to the implementation

Reply:
Resolved above.